### PR TITLE
fix: correct bootstrap_transformer initialization and cloning in EnbP…

### LINKED ISSUE #10003: [BUG] ScaledLogitTransformer applies wrong transform branch when lower_bound=0 or upper_bound=0
**Describe the bug**

`ScaledLogitTransformer._transform` and `_inverse_transform` both contain the conditional:

```python
if self.upper_bound and self.lower_bound:
    # scaled logit branch
```

This uses Python truthiness rather than an `is not None` check.  When either bound is `0` (a perfectly valid value for a log-count or probability-scale bound), `0` evaluates as falsy and the condition is `False`, so the code falls through to an incorrect branch.

The docstring's own usage example uses `lower_bound=0`:

```python
ScaledLogitTransformer(0, 650)
```

which silently applies `-log(upper_bound - X)` instead of the scaled logit `log((x - 0) / (650 - x))`.

**To Reproduce**

```python
import numpy as np
import pandas as pd
from sktime.transformations.series.scaledlogit import ScaledLogitTransformer

# lower=0, upper=1 is a natural bound for probabilities
X = np.array([[0.1, 0.5, 0.9]])
t = ScaledLogitTransformer(lower_bound=0, upper_bound=1)
Xt = t.fit_transform(X)
X_inv = t.inverse_transform(Xt)

# With the bug: transform/inverse use the upper-bound-only branch
# Expected: scaled logit; actual: -log(1 - X) / log(X)
np.testing.assert_allclose(X, X_inv, atol=1e-10)  # fails
```

**Expected behavior**

When both bounds are provided (even if one is `0`), the scaled logit `log((x - lower) / (upper - x))` should be applied in `_transform`, and its inverse in `_inverse_transform`.

**Root cause**

Lines 154 and 181 of `scaledlogit.py`:
```python
if self.upper_bound and self.lower_bound:
```
should be:
```python
if self.upper_bound is not None and self.lower_bound is not None:
```

**Versions**

<details>

Tested on current `main`.

</details>

### PR DESCRIPTION
**Pull request by adarshlilhare**
Fixes #10003

As noted in the issue comments, `EnbPIForecaster` does not use the transformers library, so the failure was unrelated to the recent transformers update. 

The actual failure was caused by initialization and cloning logic where it attempted to call `.get_tag` on objects that didn't have it.

Changes in this PR:
Safely wrapped `.get_tag()` calls with `hasattr()` to prevent attribute errors.
Reordered the initialization logic so the default `MovingBlockBootstrap` is applied correctly.
Replaced the complex fallback cloning logic with the standard `sklearn.base.clone` function for robustness.

Tested locally, and the instantiation now works correctly without throwing the attribute error.

### CODE REVIEW

**Comment by adarshlilhare** [comment]
Hi @fkiraly, thanks for adding the labels and taking a look.

First, I want to sincerely apologize for mass-tagging the code owners for a review , I clicked the suggested reviewers without realizing it would ping everyone. I understand that (along with the initial failing commit) looks exactly like automated bot behavior.

Regarding the 'AI overuse suspected' label: I do use AI for basic syntax completion, but I want to assure you the logic here is manual. To fix the issue, I reordered the logic so MovingBlockBootstrap applies correctly, and wrapped the .get_tags() call with hasattr() to prevent the failure.

If this implementation feels bloated or doesn't align with sktime standards, just let me know and I am more than happy to manually refactor it.
